app/js/lib/foxbox/recipes.js

```diff
@@ -0,0 +1,300 @@
+'use strict';
+
+// @todo Proxy recipes in the db and emit events on change (see services).
+
+// Private members.
+const p = {
+  // Private properties.
+  settings: Symbol('settings'),
+  net: Symbol('net')
+};
+
+export default class Recipe {
+  constructor(props) {
+    this[p.settings] = props.settings;
+    this[p.net] = props.net;
+  }
+
+  /**
+   * Returns a promise resolving to a list of recipes.
+   *
+   * @return {Promise}
+   */
+  getAll() {
+    return this[p.net].fetchJSON(
+      `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/services`)
+      // Get all recipes.
+      .then(services => services.filter(service =>
+        service.adapter === 'thinkerbell-adapter' &&
+        service.id !== 'thinkerbell-root-service'
+      ))
+      // Fetch their respective enabled status.
+      .then(recipes => {
+        const promises = recipes.map(recipe => {
+          const payload = [
+            { id: `${recipe.id}/get_enabled` }
+          ];
+
+          return this[p.net].fetchJSON(
+            `${this[p.net].origin}/api/v${this[p.settings].apiVersion}/` +
+            'channels/get',
+            'PUT',
+            payload);
```

Comment:
__question:__ I've noticed we use different closing bracket style, I used to put it into a separate line (as curly braces, to visually enforce block boundaries and not to keep many braces in a single place) in case arguments are wrapped and you keep it together with the last argument. Let's agree on the same style?

```
callA(
  'aaaaaaaaaaaaaaaaa',
  'bbbbbbbbbbbbbbbbb',
  callC(
     'dddddddddddddddd',
     'eeeeeeeeeeeeeeee'
  )
);

vs

callA(
  'aaaaaaaaaaaaaaaaa',
  'bbbbbbbbbbbbbbbbb',
  callC(
    'dddddddddddddddd',
    'eeeeeeeeeeeeeeee'));
```